src/main/java/network/radicle/jetbrains/radiclejetbrainsplugin/dialog/clone/CloneUtil.java

```diff
@@ -42,26 +47,47 @@ public static void doClone(@NotNull CheckoutProvider.Listener listener, @NotNull
             logger.error("Clone Failed. Destination doesn't exist");
             return ;
         }
-        var directoryName = Paths.get(clPr.directory()).getFileName().toString();
-        var parentDirectory = parent.toAbsolutePath().toString();
-        File directory = new File(parentDirectory, directoryName);
 
-        var clone = new RadClone(clPr.url(), parentDirectory);
+        try {
+            var tmpFolder = Files.createTempDirectory("project" + UUID.randomUUID());
+            tmpFolderPath = tmpFolder.toAbsolutePath().toString();
+        } catch (Exception e) {
+            BasicAction.showErrorNotification(project, RadicleBundle.message("cloneFailed"),
+                    RadicleBundle.message("tempDirError"));
+            logger.error("Unable to create temp directory");
+            return ;
+        }
+
+        var clone = new RadClone(clPr.url(), tmpFolderPath);
         var countDownLatch = new CountDownLatch(1);
+        String finalTmpFolderPath = tmpFolderPath;
         ApplicationManager.getApplication().executeOnPooledThread(() -> {
             var pr = new BasicAction(clone,project,countDownLatch).perform();
             if (pr.getExitCode() != 0) {
                 return ;
             }
             ApplicationManager.getApplication().invokeLater(() -> {
-                File oldDirectory = new File(parentDirectory, clPr.projectName(pr.getStdoutLines(true)));
-                var success = oldDirectory.renameTo(directory);
-                if (success) {
-                    listener.directoryCheckedOut(directory,null);
-                } else {
-                    listener.directoryCheckedOut(oldDirectory,null);
+                var tmpFiles = new File(finalTmpFolderPath);
+                var folders = tmpFiles.list();
+                if (folders == null || folders.length == 0) {
+                    try {
+                        FileUtils.deleteDirectory(new File(finalTmpFolderPath));
+                    } catch (IOException e) {
+                        logger.warn("unable to delete temp dir:",e);
+                    }
+                    return ;
+                }
+                var projectName = folders[0];
+                var tmpDirectory = new File(finalTmpFolderPath, projectName);
+                var selectedDirectory = new File(parent.toAbsolutePath().toString());
+                try {
+                    FileUtils.copyDirectory(tmpDirectory, selectedDirectory);
+                    FileUtils.deleteDirectory(new File(finalTmpFolderPath));
+                    listener.directoryCheckedOut(selectedDirectory,null);
```

Comment:
IIUC, you need to run in the UI thread for the listener?
If that is the case, then move only the listener code in the UI, otherwise remove whole block from `invokeLater`